tests/integration/security/file_mounted_certs/main_test.go

```diff
@@ -0,0 +1,177 @@
+//  limitations under the License.
+
+package filemountedcerts
+
+import (
+	"context"
+	"io/ioutil"
+	"path"
+	"strings"
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	inst istio.Instance
+)
+
+const (
+	PilotCertsPath = "tests/testdata/certs/pilot"
+	PilotSecretName = "test-istiod-server-cred"
+
+	ProxyMetadataJson = `
+	{
+      "PILOT_CERT_PROVIDER": "mycopki",
+      "FILE_MOUNTED_CERTS": "true",
+      "ISTIO_META_TLS_CLIENT_CERT_CHAIN": "/client-certs/cert-chain.pem",
+      "ISTIO_META_TLS_CLIENT_KEY": "/client-certs/key.pem",
+      "ISTIO_META_TLS_CLIENT_ROOT_CERT": "/client-certs/root-cert.pem",
+      "ISTIO_META_TLS_SERVER_CERT_CHAIN": "/server-certs/cert-chain.pem",
+      "ISTIO_META_TLS_SERVER_KEY": "/server-certs/key.pem",
+      "ISTIO_META_TLS_SERVER_ROOT_CERT": "/server-certs/root-cert.pem",
+	}
+`
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		Label(label.CustomSetup).
+
+		// SDS requires Kubernetes 1.13
+		RequireEnvironmentVersion("1.13").
+		Label("CustomSetup").
+		Setup(istio.Setup(&inst, setupConfig, CreateCustomIstiodSecret)).
+		Run()
+}
+
+
+func setupConfig(_ resource.Context, cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+
+	cfg.ControlPlaneValues = `
+components:
+  pilot:
+    enabled: true
+    k8s:
+      overlays:
+        - kind: Deployment
+          name: istiod
+          patches:
+            - path: spec.template.spec.containers.[name:discovery].args[1001]
+              value: "--caCertFile=/server-certs/root-cert.pem"
+            - path: spec.template.spec.containers.[name:discovery].args[1002]
+              value: "--tlsCertFile=/server-certs/cert-chain.pem"
+            - path: spec.template.spec.containers.[name:discovery].args[1003]
+              value: "--tlsKeyFile=/server-certs/key.pem"
+            - path: spec.template.spec.volumes[-1]
+              value: |-
+                name: server-certs
+                secret:
+                  secretName: ` + PilotSecretName + `
+                  defaultMode: 420
+            - path: spec.template.spec.containers[name:discovery].volumeMounts[-1]
+              value: |-
+                name: server-certs
+                mountPath: /server-certs
+
+meshConfig:
+  defaultConfig:
+    controlPlaneAuthPolicy: "MUTUAL_TLS"
+    proxyMetadata: ` + strings.Replace(ProxyMetadataJson, "\n", "", -1) +
+`
+values:
+  global:
+    pilotCertProvider: "mycopki"
+    logging:
+      level: "default:debug"
+    proxy:
+      logLevel: debug
+      componentLogLevel: "misc:debug"
+  pilot:
+    env:
+      XDS_AUTH: "false"
```

Comment:
@howardjohn , I left `XDS_AUTH: false` here because without it server/client connections to istiod will fail with the following errors:

```
2020-10-24T00:17:22.684038Z	warn	ads	Unauthorized XDS: 10.1.0.154:43728 with identity [spiffe://cluster.local/ns/mounted-certs/sa/server server.mounted-certs.svc]: no identities ([spiffe://cluster.local/ns/mounted-certs/sa/server server.mounted-certs.svc]) matched istio-fd-sds-1-77818/default
  2020-10-24T00:17:24.051879Z	warn	ads	Unauthorized XDS: 10.1.0.159:41960 with identity [spiffe://cluster.local/ns/mounted-certs/sa/client client.mounted-certs.svc]: no identities ([spiffe://cluster.local/ns/mounted-certs/sa/client client.mounted-certs.svc]) matched istio-fd-sds-1-4523/default
```

Ie. the expected identity is not in the test certificates. Please let me know if there's a way around this.

Reply:
I think thats tricky since the namespace may not be fixed. lets keep this for now, but would be good to add a comment about why we did it